chan.go

```diff
@@ -70,6 +82,20 @@ func (r *chanResponse) Length() uint64 {
 	return r.length
 }
 
+func (re *chanResponse) Head() Head {
+	<-re.wait
+
+	err, ok := re.err.(*cmdkit.Error)
```

Comment:
This will treat EOF as an error. Maybe we should just have a separate closed flag if we need that and not write EOF to the error field.

Reply:
Actually, we can just check `<-re.closed`.

Reply:
You are right that this is not correct. However, there can be an error even if `re.closed` already is closed, so I don't think it's a good indicator. I think I fixed this locally and will push the changes once I'm through you first pass.